skrifa/src/meta/metrics.rs

```diff
@@ -0,0 +1,434 @@
+//! Global font and glyph specific metrics.
+
+use read_fonts::{
+    tables::{glyf::Glyf, hmtx::LongMetric, hvar::Hvar, loca::Loca},
+    types::{BigEndian, GlyphId},
+    TableProvider,
+};
+
+use crate::NormalizedCoord;
+
+/// Metrics for a text decoration.
+#[derive(Copy, Clone, PartialEq, Default, Debug)]
+pub struct Decoration {
+    /// Offset to the top of the decoration from the baseline.
+    pub offset: f32,
+    /// Thickness of the decoration.
+    pub thickness: f32,
+}
+
+/// Extents of a rectangular region.
+#[derive(Copy, Clone, PartialEq, Default, Debug)]
+pub struct BoundingBox {
+    /// Minimum x-coordinate.
+    pub x_min: f32,
+    /// Minimum y-coordinate.
+    pub y_min: f32,
+    /// Maximum x-coordinate.
+    pub x_max: f32,
+    /// Maximum y-coordinate.
+    pub y_max: f32,
+}
+
+/// Metrics that apply to all glyphs in a font.
+#[derive(Copy, Clone, PartialEq, Default, Debug)]
+pub struct Metrics {
+    /// Number of font design units per em unit.
+    pub units_per_em: u16,
+    /// Number of glyphs in the font.
+    pub glyph_count: u16,
+    /// True if the font is not proportionally spaced.
+    pub is_monospace: bool,
+    /// Italic angle in counter-clockwise degrees from the vertical. Zero for upright text,
+    /// negative for text that leans to the right.
+    pub italic_angle: f32,
+    /// Distance from the baseline to the top of the alignment box.
+    pub ascent: f32,
+    /// Distance from the baseline to the bottom of the alignment box.
+    pub descent: f32,
+    /// Recommended additional spacing between lines.
+    pub leading: f32,
+    /// Distance from the baseline to the top of a typical English capital.
+    pub cap_height: Option<f32>,
+    /// Distance from the baseline to the top of the lowercase "x" or
+    /// similar character.
+    pub x_height: Option<f32>,
+    /// Average width of all non-zero width characters in the font.
+    pub average_width: Option<f32>,
+    /// Maximum advance width of all characters in the font.
+    pub max_width: Option<f32>,
+    /// Metrics for an underline decoration.
+    pub underline: Option<Decoration>,
+    /// Metrics for a strikeout decoration.
+    pub strikeout: Option<Decoration>,
+    /// Union of minimum and maximum extents for all glyphs in the font.
+    pub bounds: Option<BoundingBox>,
+}
+
+impl Metrics {
+    /// Creates new metrics for the given font, size in pixels per em, and
+    /// normalized variation coordinates.
+    ///
+    /// If `size` is `None` or `Some(0.0)`, resulting metric values will be in font units.
+    pub fn from_font<'a>(
+        font: &impl TableProvider<'a>,
+        size: Option<f32>,
```

Comment:
A new user is likely to be mystified as to what size value to pass. What guidance for typical use can we provide?

Reply:
Since all “size” values in this crate should be in ppem, I can add a section on sizing to the crate root doc and link to it from each method that makes use.

Reply:
Added a new `Size` type that is better documented.